### PR TITLE
Mastodon (toot) capability integration

### LINKED ISSUE #8: feature-request: Notifications to mastodon
I currently run two separate github event notification systems in parallel:

- one based on an old version of [bitcoinacks](https://github.com/PierreRochard/bitcoin-acks)—this can do notifications to Mastodon and Twitter (I only care about the former right now), for merges (polling based)
- `ghi`—for notifying IRC (webhook notification based)

It would be nice to have this in one system, specified by one configuration file.

For comparison here was the patch to add "toot" support to `bitcoinacks`: https://github.com/PierreRochard/bitcoin-acks/commit/9f828d960fd499a2881779156b8adf260efbfe0e

### PR DESCRIPTION
**Pull request by Kvaciral**
To support Mastodon as an outlet for ghi I:
- edited ghi/configuration.py to read, check and use the (optional) Mastodon-settings from the .yml
- created ghi/ghimastodon.py to facilitate creating credential-files, logging onto a Mastadon instance and sending messages (toots) to it by using the Mastadon.py module
- changed some naming and mastodon-specific things in most of the files
- changed the README.md en .ghi.yml.example to include the new Mastodon-support
- added relevant exception-handlers

No changes to the config-file are necessary when using IRC. The only notable change is the addition of "IRC - " in the log-messages when (dis)connecting and sending messages to IRC.

It works quite well with server.py, haven't tested it with AWS though, but should work as well.

Closes #8

### CODE REVIEW

**Comment by laanwj** [comment]
Thanks !  I have tested the mastodon functionality (in addition to IRC functionality) and it works as documented.

**Comment by Kvaciral** [comment]
My patch has been used for a couple of days, also with a complex configfile. So it has been tested quite extensively. Works like a charm!!

File: README.md [hunk 1]
```diff
@@ -191,14 +234,24 @@ pools: # required
...
       password: abc123 # optional, but if it's required by the nick it must be set here or with environment variable
       channels: # required
         - my-cool-channel # at least 1 channel is required
+    mastodon: # required if Mastodon is one of the outlets
+      instance: https://mstdn.social # required
+      user: happy@place.net # required
+      password: myBotPassword123! # required
+      secretspath: /home/thatsme/my/secrets/ # required
+      appname: my-mastodon-bot # required
+      merges_only: # optional, default is true
```
Comment (gkrizek) [inline README.md:251]:
```suggestion
      merges_only: true # optional, default is true
```

**Review by gkrizek — COMMENTED** [review]
This is great! Thank you for creating the PR. It will be awesome to have Mastodon support. Had one tiny nit in the README, but otherwise I think this looks good.

One clarifying question before merging; If an existing user is running this for IRC and they upgrade, will it work the same as it did before? I know you said in your description that no changes are needed in the config, just making sure this is a 'drop in' upgrade and doesn't require any changes for users.

File: README.md (hunk 1 above)
Comment (Kvaciral) [reply README.md]: Oops, nice catch.. I've missed that one. That'd be an TypeError if used like that in an actual configfile.

Yes, existing users who only want to use it for IRC won't have to change a thing for the same functionality. The only difference for them is the added "IRC - " in some of the log-messages.

**Comment by Kvaciral** [comment]
I don't have the know-how to test the Mastodon-functionality with AWS, but I think it'll work? (not sure about the creation and placement of the credential files)

**Comment by gkrizek** [comment]
Perfect. This looks good! Thanks for making the PR.

**Review by gkrizek — APPROVED** [review]

**Comment by Kvaciral** [comment]
You're welcome, thanks for the merge! :D

**Pull request merged**